src/snowflake/connector/file_transfer_agent.py

```diff
@@ -183,6 +189,35 @@ def _update_progress(
     return progress == 1.0
 
 
+def _chunk_size_calculator(file_size: int) -> int:
+    # S3 has limitation on the num of parts to be uploaded, this helper method recalculate the num of parts
+    if file_size > S3_MAX_OBJECT_SIZE:
+        # check if we don't exceed the allowed S3 max file size 5 TiB
+        error_message = f"File size {file_size} exceeds the maximum file size {S3_MAX_OBJECT_SIZE} allowed in S3."
+        logger.error(error_message)
+        Error.errorhandler_wrapper(
```

Comment:
What Exception is this raising?

Reply:
is it raising the most generic error `Error` defined in line 201.
I don't find an error type that best fits this case, so just putting a generic one, we can improve in the future
wdyt